### PR TITLE
Cleanup skipped parser tests

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Tests Added + Pass?      | Yes
| License                  | MIT

In this PR we clean up some previously skipped parser tests:
- Tests related to regex syntax errors are removed because Babel does not parse regex, nor do we intend to do so in the future
- Tests that we already supported are now enabled

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/15903"></a>

### CODE REVIEW

File: packages/babel-parser/test/fixtures/core/categorized/invalid-object-setter-no-param/output.json [hunk 1]
```diff
@@ -0,0 +1,52 @@
+{
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/test/fixtures/core/categorized/invalid-object-setter-no-param/output.json:1]: This is also tested in https://github.com/babel/babel/blob/18d13d0032825b6d690b81f9d7e1d5a22ec97b30/packages/babel-parser/test/fixtures/core/object/invalid-setter/input.js

**Review by nicolo-ribaudo — APPROVED** [review]
This PR is incredibly difficult to review, but given that it's just adding tests and deleting disabled tests, rubber stamp LGTM.

File: packages/babel-parser/test/fixtures/core/categorized/invalid-object-setter-no-param/output.json (hunk 1 above)
Comment (JLHwung) [reply packages/babel-parser/test/fixtures/core/categorized/invalid-object-setter-no-param/output.json:1]: Good catch! I will check if the other accessor params test already have duplicates, if so I tend to just remove them.

**Comment by nicolo-ribaudo** [comment]
Let's merge this, we can always iterate in follow-up PRs if we spend other time improving/categorizing our parser tests :)

**Pull request merged**